ci/functional/node_checkout.sh

```diff
@@ -0,0 +1,211 @@
+
+daos_rpms_clients=("daos-client" "daos" "libfabric" "mercury" "protobuf-c")
+
+if [[ $1 == "server" ]]; then
+    rpms_list=("${prereq_rpms_list[@]}" "${daos_rpms_servers[@]}")
+elif [[ $1 == "client" ]]; then
+    rpms_list=("${prereq_rpms_list[@]}" "${daos_rpms_clients[@]}")
+else
+    rpms_list=("${prereq_rpms_list[@]}")
+fi
+
+echo "Checking for following list of rpms: ${rpms_list[@]}"
+
+declare -a rpms_installed
+declare -a rpms_not_installed
+
+for i in "${!rpms_list[@]}"; do
+    rc_rpm=`rpm -qa | grep -q ${rpms_list[i]};echo $?`
+    if [ "${rc_rpm}" != 0 ]; then
+        rpms_not_installed=(${rpms_not_installed[@]} "${rpms_list[i]}")
+    else
+        rpms_installed=(${rpms_installed[@]} "${rpms_list[i]}")
+    fi
+done
+
+echo "Installed rpms: ${rpms_installed[@]}"
+
+if [ ${#rpms_not_installed[@]} != 0 ]; then
+    echo "[FAILED] ${rpms_not_installed[@]} not installed"
+    exit $rc
+fi
+
+}
+
+# Verify with lscpu that VT-x is enabled
+vtx_enabled(){
+
+output=`lscpu | grep Virtualization`
+awk '$0~/VT-x/{print "VT-x is enabled\n"}' <<< $output
+
+}
+
+# Verify that the number of NUMA nodes is equal to the number of sockets
+check_numa_sockets(){
+
+output=`lscpu | egrep -e Socket -e NUMA`
+num_sockets=$(echo $output | cut -d ' ' -f 2)
+num_numa_nodes=$(echo $output | cut -d ' ' -f 5)
+if [ "$num_sockets" -eq "$num_numa_nodes" ]; then
+    echo "Number of sockets is equivalent to number of numa nodes"
+    echo "Number of sockets: ${num_sockets}"
+    echo -e "Number of Numa Nodes: ${num_numa_nodes}\n"
+else
+    echo "[FAILED] Number of sockets is not equal to number of numa nodes"
+    exit $rc
+fi
+
+}
+
+
+# Verify valid IB state
+verify_ibstate(){
+
+output=`ibstatus | egrep -e state`
+num_active_ports=$(echo $output | awk -F': ACTIVE' 'NF{print NF-1}')
+if [ "$num_active_ports" -gt 0 ]; then
+    echo -e "Number of active ib ports: ${num_active_ports}\n"
+else
+    echo "[FAILED] No active ib port"
+    exit $rc
+fi
+
+}
+
+# Verify all the IB interfaces have IP addresses attached
+verify_ip_for_interfaces(){
+
+# get all available interfaces
+available_ifaces=`ls /sys/class/net`
+echo $available_ifaces
+iface_count=$(echo $available_ifaces | awk -F' ' 'NF{print NF-1}')
+echo "Total number of interfaces: ${iface_count}"
+
+output=`ip -4 -o a | cut -d ' ' -f 2,7 | cut -d '/' -f 1`
+echo $output
+set -f
+array=(${available_ifaces// / })
+declare -a iface_with_ip
+declare -a iface_without_ip
+
+for i in "${!array[@]}";do
+    if [[ $output =~ "${array[i]}" ]]; then
+        echo "IP available for interface: ${array[i]}"
+        iface_with_ip=(${iface_with_ip[@]} "${array[i]}")
+    else
+        echo "IP not available for interface: ${array[i]}"
+        iface_without_ip=(${iface_without_ip[@]} "${array[i]}")
+    fi
+done
+
+pattern='ib'
+if [[ "${iface_without_ip[@]}" =~ "${pattern}" ]]; then
```

Comment:
(lint) Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @). [SC2199]
(lint) Remove quotes from right-hand side of =~ to match as a regex rather than literally. [SC2076]